src/02-common-principles.md

```diff
@@ -303,6 +303,30 @@ sub- prefix is not part of the subject label, but must be included in file names
 MUST be integers. Those labels MAY include zero padding, but this is NOT
 RECOMMENDED to maintain their uniqueness.
 
+## The contrast suffix
+
+In addition to the `*<label_identifier>-<custom_label>*` pairs (e.g. `sub-001`),
+it is RECOMMENDED that file names contain a suffix of the form `*_<suffix>.`,
+which directly precedes the file format delimiter. The string values of this
+suffix indicate  the contrast of the file in question.
+
+This specification is often needed, as the NIfTI header does not allow us to
+specify what unit or contrast the voxel values correspond to. The decision of
+BIDS to track this information in the file name rather than in the JSON sidecar
+file, is based on (a) the high relevance of this field for rapid file selection
+(b) the low number but large scope of brain imaging contrasts. For non-NIfTI
+files, this suffix provides more user-friendly recognition of the feature
+documented by the file. A file ending in `_events.tsv` is more unambiguously
+understood as documenting the events than if this information had to be inferred
+purely from the `.tsv` format.
+
+It is particularly noteworthy that there can only be one contrast suffix per file,
+as the voxel values can only correspond to one contrast. This field is not
+designed to track historical information, and the suffix `_cbv_tstat.*` is
+nonsensical. If multiple contrasts (e.g. `_cbv.*` and `_bold.*`) are acquired in
+one session, the corresponding files MUST be differentiated by other labels
+(e.g. `*run-<label>`).
```

Comment:
This new rule breaks a lot of existing datasets and I am not convinced it is necessary. I would recommend removing it.

Reply:
How then would you adapt the suffix in the described use case?

Reply:
Let me clarify - right now it's allowed to have `sub-01/anat/sub-01_T1w.nii.gz` and `sub-01/anat/sub-01_T2w.nii.gz` in the same dataset. The rule you added in this PR would require adding `run` or other keywords to those filenames. Was this your intention?

Reply:
I see, my concern was the following. Let's say you have:

```
sub-01/anat/sub-01_bold.nii.gz
sub-01/anat/sub-01_cbv.nii.gz
```

And you perform some statistic, meaning that the values in your NIfTI will no longer code for bold or cbv but for a t-statistic. Concatenating the fileds would be confusing since (unlike subject, session, etc.) they don't have a prefix label to identify what they are. Furthermore, only the last one would actually specify what the values are representing. A t-statistic is a t-statistic regardless of the nature of the data it was applied to.

So you would need the uniquely identified run:

```
sub-01/anat/sub-01_run-0_bold.nii.gz
sub-01/anat/sub-01_run-1_cbv.nii.gz
```

to arrive at an unambiguous statistic representation:

```
sub-01/anat/sub-01_run-0_tstat.nii.gz
sub-01/anat/sub-01_run-1_tstat.nii.gz
```

Reply:
I see your concern now. This filename conflict on the level of derivatives, can be resolved by a combination of `desc` filename keyword and `RawSource` JSON keyword. See https://github.com/bids-standard/bids-specification/pull/109 for details.

Reply:
hm, I get that you probably put a lot of thought in to this, but `desc` sounds like yet another ad-hoc field which can contain whatever and be overall confusing and limit reusability of workflows because there's no way you can use this key and predict what it might hold. Sounds like mitigating the suffix problem by duplicating it :( Are you totally sold on this?

Reply:
This discussion would be better placed on https://github.com/bids-standard/bids-specification/pull/109